src/classifier/SparseLabeledParameterVector.h

```diff
@@ -98,6 +103,34 @@ class SparseLabelWeights : public LabelWeights {
     }
     label_weights_.push_back(std::pair<int, double>(label, weight));
   }
+  // Sets new weight, normalize it and returns previous value of label_weights_[k].second, with k such that label == label_weights_[k].first
+  double SetWeightAndNormalize(int label, double value, double scaling_factor) {
```

Comment:
Call this scale_factor instead of scaling_factor, for consistency with the already existing member variable scale_factor_.